daemon/src/model/cfd.rs

```diff
@@ -2416,6 +2416,83 @@ mod tests {
         assert_eq!(no_rollover_reason, NoRolloverReason::Closed);
     }
 
+    /// Cover scenario where trigger a collab settlement during ongoing rollover
+    ///
+    /// In this scenario the collab settlement finished before the rollover finished.
+    /// If we trigger a collaborative settlement during rollover, the settlement will have priority
+    /// over the rollover. Upon finishing the rollover we fail because the cfd was already
+    /// settled.
+    #[test]
+    fn given_collab_settlement_then_cannot_finish_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok());
+
+        let (cfd, _, _, _) = cfd
+            .with_lock(taker_keys, maker_keys)
+            .dummy_collab_settlement_taker(opening_price);
+
```

Comment:
should we test with the ongoing collab settlement here? (or maybe in yet another test?) I have to admit still am pretty hazy in regards to the distinction "in collab settlement" vs "in ongoing collab settlement".

is this mapping right?
- "in collab settlement" - collab settlement proposed
- "in ongoing collab settlement" -  collab settlement got accepted and we're doing the protocol

Reply:
The test below covers the `ongoing` case. I wanted to always cover both collab settlement finished (and we have the tx) and ongoing (where we have the settlement only). It's quite a lot of boilerplate which is annoying.

As for the mapping

> "in collab settlement" - collab settlement proposed
> "in ongoing collab settlement" - collab settlement got accepted and we're doing the protocol

that's not how I defined it, my distinction is:

1. ongoing: we have a proposal but no collab settlement transaction yet (i.e. the settlement protocol is not finished yet)
2. we have the collab settlement transaction (the settlement protocol is finished) but we don't know anything about finality yet
3. we reached finality of the transaction